### PR TITLE
Add DEBUG_DNSMASQ_LINES debug flag

### PR DESCRIPTION
**Pull request by DL6ER**
Signed-off-by: DL6ER <dl6er@dl6er.de>

**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

## 10

---

This PR adds a new debug option. Then enabled, the embedded `dnsmasq` will append `file:line` information for each line (see example below).

`DEBUG_DNSMASQ_LINES=false` (standard):
```
Dec 10 10:53:31 dnsmasq[21723]: query[A] team.pi-hole.net from 192.168.2.223
Dec 10 10:53:31 dnsmasq[21723]: forwarded team.pi-hole.net to 8.8.8.8
Dec 10 10:53:31 dnsmasq[21723]: query[AAAA] team.pi-hole.net from 192.168.2.223
Dec 10 10:53:31 dnsmasq[21723]: forwarded team.pi-hole.net to 8.8.8.8
Dec 10 10:53:31 dnsmasq[21723]: reply team.pi-hole.net is 46.4.71.158
Dec 10 10:53:31 dnsmasq[21723]: reply team.pi-hole.net is 2a01:4f8:140:329e::1:4
```

`DEBUG_DNSMASQ_LINES=true`:
```
Dec 10 10:55:12 dnsmasq[21723]: query[A] team.pi-hole.net from 192.168.2.223 (src/dnsmasq/forward.c:1558)
Dec 10 10:55:12 dnsmasq[21723]: forwarded team.pi-hole.net to 8.8.8.8 (src/dnsmasq/forward.c:556)
Dec 10 10:55:12 dnsmasq[21723]: query[AAAA] team.pi-hole.net from 192.168.2.223 (src/dnsmasq/forward.c:1558)
Dec 10 10:55:12 dnsmasq[21723]: forwarded team.pi-hole.net to 8.8.8.8 (src/dnsmasq/forward.c:556)
Dec 10 10:55:12 dnsmasq[21723]: reply team.pi-hole.net is 46.4.71.158 (src/dnsmasq/cache.c:481)
Dec 10 10:55:12 dnsmasq[21723]: reply team.pi-hole.net is 2a01:4f8:140:329e::1:4 (src/dnsmasq/cache.c:481)
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,7 +95,7 @@ else
   EXTRAWARN_GCC8=
 endif
 EXTRAWARN=-Werror -Waddress -Wlogical-op -Wmissing-field-initializers -Woverlength-strings -Wformat -Wformat-nonliteral -Wuninitialized -Wswitch-enum -Wshadow \
--Wfloat-equal -Wunsafe-loop-optimizations -funsafe-loop-optimizations -Wbad-function-cast -Wwrite-strings -Wparentheses -Wlogical-op -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Winline $(EXTRAWARN_GCC8)
+-Wfloat-equal -Wbad-function-cast -Wwrite-strings -Wparentheses -Wlogical-op -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Winline $(EXTRAWARN_GCC8)
```
Comment (Mcat12) [inline Makefile]: Why are these removed?

Comment (PromoFaux) [reply Makefile]: They're unnecessary. They were removed in `new/http` some time ago.
